tests/framework/utils.py

```diff
@@ -261,6 +261,60 @@ def __str__(self):
         return str(self._inner)
 
 
+class ExceptionAggregator(Exception):
+    r"""
+    Abstraction over an exception with message formatter.
+
+    The failure is formatted as a list of elements with the following
+    structure:
+    ```
+    [title1]\n
+    [message11]\n
+    [message12]\n
+    [message13]\n
+    ...
+    [title2]\n
+    [message21]\n
+    [message22]\n
+    [message23]\n
+    ...
+    ...
+    ```
+    """
+
+    def __init__(self, hit_return=False):
+        """Initialize the exception aggregator."""
+        super().__init__()
+        self._failures = list()
+
+        # If `hit_return` is True then the failures will start one row below,
+        # in the logs. This is useful for having the failures starting on an
+        # empty line, keeping the formatting nice and clean.
```

Comment:
I think that this parameter can introduce confusion, since the naming is not clear.

Seeing the tiny benefit it adds, I would remove it completely and just add a default. I don't have strong feelings about it, but if we keep it, let's rename it to smth like `add_newline`